bin/test-storybook.js

```diff
@@ -34,32 +51,69 @@ function sanitizeURL(url) {
   return finalURL;
 }
 
-const targetURL = sanitizeURL(process.env.TARGET_URL || `http://localhost:6006`);
+async function executeJestPlaywright(args) {
+  const jest = require('jest');
+  let argv = args.slice(2);
+
+  const jestConfigPath = fs.existsSync('test-runner-jest.config.js')
+    ? 'test-runner-jest.config.js'
+    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js');
+
+  argv.push('--config', jestConfigPath);
+
+  await jest.run(argv);
+}
 
-urlExists(targetURL, function (err, exists) {
-  if (!exists) {
-    console.error(`[test-storybook] It seems that your Storybook instance is not running at: ${targetURL}. Are you sure it's running?`)
-    process.exit(1)
+async function checkStorybook(url) {
+  try {
+    const res = await fetch(url, { method: 'HEAD' });
+    if (res.status !== 200) throw new Error(`Unxpected status: ${res.status}`);
+  } catch (e) {
+    console.error(
+      `[test-storybook] It seems that your Storybook instance is not running at: ${url}. Are you sure it's running?`
+    );
+    process.exit(1);
   }
+}
 
-  executeJestPlaywright()
-});
+async function fetchStoriesJson(url) {
+  const storiesJsonUrl = new URL('stories.json', url).toString();
+  let tmpDir;
+  try {
+    const res = await fetch(storiesJsonUrl);
+    const json = await res.text();
+    const titleIdToTest = transformPlaywrightJson(json);
 
-function executeJestPlaywright() {
-  const fs = require('fs');
-  const path = require('path');
+    tmpDir = tempy.directory();
+    Object.entries(titleIdToTest).forEach(([titleId, test]) => {
+      const tmpFile = path.join(tmpDir, `${titleId}.test.js`);
+      fs.writeFileSync(tmpFile, test);
+    });
+  } catch (err) {
+    console.error(`[test-storybook] Failed to fetch stories.json from ${storiesJsonUrl}.`);
+    console.error(
+      'More info: https://github.com/storybookjs/test-runner/blob/main/README.md#storiesjson-mode'
```

Comment:
Nice! Just FYI this is how it looks like when it errors out:

![image](https://user-images.githubusercontent.com/1671563/151410668-f14484a6-4f74-4d5a-b20f-ab9ccd3e38c6.png)